Source/ConversationV1/Tests/ConversationTests.swift

```diff
@@ -0,0 +1,215 @@
+
+    // timeout for asynchronous completion handlers
+    private let timeout: NSTimeInterval = 30.0
+
+    // MARK: - Conversation Tests
+
+    /// Test the Conversation service by executing each operation with valid parameters.
+    func testConversation() {
+
+        // Send a test message with no history/context
+        sendInitialMessageNoContext()
+
+        // Send a test message with no history/context, and then send a follow-up message
+        sendMessageWithContext()
+        
+        // verify error conditions
+        executeNegativeTests()
+    }
+
+    /// Test the Conversation service by executing each operation with invalid parameters.
+    func executeNegativeTests() {
+
+    }
+
+    // MARK: - Helper functions
+
+    /// Load credentials and instantiate Dialog service
+    override func setUp() {
+        super.setUp()
+
+        // identify credentials file
+        let bundle = NSBundle(forClass: self.dynamicType)
+        guard let url = bundle.pathForResource("Credentials", ofType: "plist") else {
+            XCTFail("Unable to locate credentials file.")
+            return
+        }
+
+        // load credentials from file
+        let dict = NSDictionary(contentsOfFile: url)
+        guard let credentials = dict as? Dictionary<String, String> else {
+            XCTFail("Unable to read credentials file.")
+            return
+        }
+
+        // read Dialog username
+        guard let username = credentials["ConversationUsername"] else {
+            XCTFail("Unable to read Conversation username.")
+            return
+        }
+
+        // read Dialog password
+        guard let password = credentials["ConversationPassword"] else {
+            XCTFail("Unable to read Conversation password.")
+            return
+        }
+
+        // instantiate the service
+        if service == nil {
+            service = Conversation(username: username, password: password)
+        }
+    }
+
+    /// Wait for an expectation to be fulfilled.
+    func waitForExpectation() {
+        waitForExpectationsWithTimeout(timeout) { error in
+            XCTAssertNil(error, "Timeout.")
+        }
+    }
+
+    // MARK: - Positive tests
+
+    /// Create the Dialog application
+    func sendInitialMessageNoContext() {
+        let description = "Sending first message with no context."
+        let expectation = expectationWithDescription(description)
+
+        sendInitialMessage() {
+            (response) in
+                expectation.fulfill()
+            }
+
+        waitForExpectation()
+    }
+    
+    /// Create the Dialog application
+    func sendMessageWithContext() {
+        let description = "Sending first message with no context."
+        let expectation = expectationWithDescription(description)
+        
+        sendInitialMessage() {
+            (response) in
+            self.sendFollowupMessage(response) {
+                (followupResponse) in
```

Comment:
Again, it would be great to check some of the properties in the `followupResponse` object to make sure that you're getting the response that you expect.